From afd5eab62b0a55e9caf097ec5f6cb0b954a5db7f Mon Sep 17 00:00:00 2001
From: Ren-Pei Zeng <kamesan@chromium.org>
Date: Mon, 7 Nov 2022 20:26:45 -0800
Subject: [PATCH] base/trace_event: Fix unused function template error

Remove "static" from template function in the header since it can cause
Wunused-template errors when including it.

Bug: b:256708604
Test: Build targets with ChromeOS libchrome.
Change-Id: I50ff5b378ccc649d969ad1997efd18c0af4de14f
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3995688
Reviewed-by: Daniel Cheng <dcheng@chromium.org>
Auto-Submit: Ren-Pei Zeng <kamesan@chromium.org>
Commit-Queue: Ren-Pei Zeng <kamesan@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1068482}


CrOS-Libchrome-Original-Commit: 6727ac5ff3c692ded993eeed1f3bec49fb39fdf9
(cherry picked from commit 3c8e3fb018f56cc43c45bced20c3305f18510d97)
---
 base/trace_event/trace_event.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/base/trace_event/trace_event.h b/base/trace_event/trace_event.h
index 02f96b1ec7..e1f29d8c7b 100644
--- a/base/trace_event/trace_event.h
+++ b/base/trace_event/trace_event.h
@@ -697,10 +697,10 @@ inline base::trace_event::TraceEventHandle AddTraceEvent(
 }
 
 template <class ARG1_TYPE>
-static void AddMetadataEvent(const unsigned char* category_group_enabled,
-                             const char* event_name,
-                             const char* arg_name,
-                             ARG1_TYPE&& arg_val) {
+void AddMetadataEvent(const unsigned char* category_group_enabled,
+                      const char* event_name,
+                      const char* arg_name,
+                      ARG1_TYPE&& arg_val) {
   base::trace_event::TraceArguments args(arg_name,
                                          std::forward<ARG1_TYPE>(arg_val));
   trace_event_internal::AddMetadataEvent(category_group_enabled, event_name,
-- 
2.38.1.431.g37b22c650d-goog

